Stellar-transaction.x

```diff
@@ -474,23 +474,79 @@ struct LiquidityPoolWithdrawOp
     int64 minAmountB; // minimum amount of second asset to withdraw
 };
 
-enum HostFunction
+enum HostFunctionType
 {
-    HOST_FN_INVOKE_CONTRACT = 0,
-    HOST_FN_CREATE_CONTRACT_WITH_ED25519 = 1,
-    HOST_FN_CREATE_CONTRACT_WITH_SOURCE_ACCOUNT = 2,
-    HOST_FN_CREATE_TOKEN_CONTRACT_WITH_SOURCE_ACCOUNT = 3,
-    HOST_FN_CREATE_TOKEN_CONTRACT_WITH_ASSET = 4
+    HOST_FUNCTION_TYPE_INVOKE_CONTRACT = 0,
+    HOST_FUNCTION_TYPE_CREATE_CONTRACT = 1,
```

Comment:
If I understand this correctly, we can call these three "host functions" from `InvokeHostFunctionOp`, but contracts won't necessarily have the same access. For example, it doesn't sound like contracts have access to `HOST_FUNCTION_TYPE_CREATE_CONTRACT`. Does the naming here make sense then? Should `HOST_FUNCTION` still be used here?

Reply:
The invoke-host-function-invoke-contract-function nested concept is not particularly fun to explain to folks either in person, or in docs.

I don't feel strongly about where this goes, but I suspect from a developer experience point-of-view if we are adding XDR types – which is attractive – we could considering having separate operations for each Soroban operation, which falls in line with historically what we've done, and has some benefits for ingesting applications that categorize based on operation type (this isn't something that needs to drive any decision, just noting alignment).

Reply:
We could just remove the `HOST_FUNCTION_TYPE*`. You've already done a good job of abstracting away the host function details by no longer requiring the user to pass the exact `ScVec` parameters that get passed to the host function. So why even keep the concept of host functions here?

Separate operations is an interesting idea, and one that would be easy to do. One of the advantages of a single op with different options is that we don't need to plumb each new operation through core as we iterate. This is not that big of a deal, so I'm not opposed to separate operations if we see the benefit.

Reply:
Separate ops is something to consider before V1, but that's a relatively significant change from the core standpoint (in terms of diffs etc, not necessarily in terms of complexity). I would propose to not try doing that within this PR to keep the reviews focused on one thing.

Reply:
I agree - separate ops should not be part of this PR and require further discussion.

Reply:
I would actually suggest to also do the renaming separately for the same reason (unless we only rename the enum; but then the op name itself will be confusing). There is a significant amount of diffs in core/host related only to the deployment changes and I don't want to additionally mix it with no-op renaming.

Reply:
Makes sense to me.